pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableTenantConfigs.java

```diff
@@ -46,15 +47,22 @@
   @Produces(MediaType.APPLICATION_JSON)
   @Path("/tables/{tableName}/rebuildBrokerResourceFromHelixTags")
   @ApiOperation(value = "Rebuild broker resource for table", notes = "when new brokers are added")
-  @ApiResponses(value = {@ApiResponse(code=200, message = "Success"),
-      @ApiResponse(code = 404, message = "Table not found"),
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 400, message = "Bad request: table name has to be with table type"),
       @ApiResponse(code = 500, message = "Internal error rebuilding broker resource or serializing response")})
-  public String rebuildBrokerResource(
+  public SuccessResponse rebuildBrokerResource(
       @ApiParam(value = "Table name (with type)", required = true) @PathParam("tableName") String tableNameWithType
-  ) throws JSONException {
-    final PinotResourceManagerResponse pinotResourceManagerResponse =
-        _helixResourceManager.rebuildBrokerResourceFromHelixTags(tableNameWithType);
-
-    return pinotResourceManagerResponse.toJSON().toString();
+  ) {
+    if (TableNameBuilder.extractRawTableName(tableNameWithType).equals(tableNameWithType)) {
```

Comment:
I don't think we need this check since you are already checking `tableConfig == null` within `rebuildBrokerResourceFromHelixTags()`

Reply:
But think it this way: whatever mistakes client makes we'll return 400, whatever the mistakes server has we'll return 500. Obviously if client gives a wrong table name, we shouldn't return a 500 since it's not a mistake from the server.

Reply:
You can catch `InvalidConfigException` and throw 400 error since invalid config exception indicates there's something wrong with the user input.